src/main/scala/org/constellation/consensus/Round.scala

```diff
@@ -545,6 +607,13 @@ class Round(
     }
   }
 
+  def setRoundStage(stage: RoundStage): Unit =
+    roundStage = stage
+
+  def verifyRoundStage(forbiddenStages: Set[RoundStage]): Unit =
+    if (forbiddenStages.contains(roundStage)) {
+      throw new RuntimeException(s"Received message from previous round stage. Current round stage is $roundStage")
```

Comment:
There is no guarantee of message ordering so we should probably discard those message instead of throwing errors

Reply:
Do we really need to process the consensus round if there is something wrong? I'd keep it as it is to close the round and start another one. Later on, we can change this implementation. WDYT @wkoszycki ?